osmoutils/binary_search.go

```diff
@@ -0,0 +1,95 @@
+package osmoutils
```

Comment:
What's the standard for methods that go into `osmoutils` package and methods that stay within a specific package?

Reply:
In this case, they could plausibly belong in osmomath